flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/DefaultConfigurableOptionsFactory.java

```diff
@@ -65,7 +65,13 @@
  * An implementation of {@link ConfigurableRocksDBOptionsFactory} using options provided by {@link
  * RocksDBConfigurableOptions}. It acts as the default options factory within {@link
  * EmbeddedRocksDBStateBackend} if the user did not define a {@link RocksDBOptionsFactory}.
+ *
+ * <p>After FLINK-24046, we refactor the config procedure for RocksDB. User could use {@link
+ * ConfigurableRocksDBOptionsFactory} to apply some customized options. Besides this, we load the
+ * configurable options in {@link RocksDBResourceContainer} instead of {@link
+ * DefaultConfigurableOptionsFactory}. Thus, we mark this factory Deprecated.
```

Comment:
I think we should tell user explictly that this class is ignored for general case and still kept for backward compatibility if user still leverage this class.